conda_build/conda_interface.py

```diff
@@ -135,3 +136,160 @@ def which_prefix(path):
             # we cannot chop off any more directories, so we didn't find it
             return None
         prefix = dirname(prefix)
+
+if parse_version(conda.__version__) >= parse_version("4.3"):
+    from conda.exports import FileMode, NodeType
+    FileMode, PathType = FileMode, NodeType
+    from conda.export import EntityEncoder
+    EntityEncoder = EntityEncoder
+    from conda.export import CrossPlatformStLink
+    CrossPlatformStLink = CrossPlatformStLink
+else:
+    from json import JSONEncoder
+    from os import lstat
+    import os
+
+    class PathType(Enum):
+        """
+        Refers to if the file in question is hard linked or soft linked. Originally designed to be used
+        in paths.json
+        """
+        hardlink = "hardlink"
+        softlink = "softlink"
+
+        @classmethod
+        def __call__(cls, value, *args, **kwargs):
+            if isinstance(cls, value, *args, **kwargs):
```

Comment:
Please test these member functions or remove them if they are not used in conda-build.